rllib/core/rl_trainer/rl_trainer.py

```diff
@@ -216,14 +217,32 @@ def compile_results(
         Returns:
             A dictionary of results.
         """
+        # TODO (Kourosh): This method assumes that all the modules with in the
+        # marl_module are accessible via looping through it rl_modules. This may not be
+        # true for centralized critic for example. Therefore we need a better
+        # generalization of this base-class implementation.
         loss_numpy = convert_to_numpy(postprocessed_loss)
-        rewards = batch["rewards"]
-        rewards = convert_to_numpy(rewards)
-        return {
-            "avg_reward": rewards.mean(),
-            **loss_numpy,
+        batch = convert_to_numpy(batch)
```

Comment:
why do you need to convert batch to numpy?

Reply:
We always report results as numpy as these will have to go "through ray" (serialization) anyways in case we run e.g. tune trials with this trainer.

Reply:
Yeah I hope we don’t report batches as stats :)